security/firewall_restriction.rst

```diff
@@ -4,19 +4,77 @@
 How to Restrict Firewalls to a Request
 ======================================
 
-When using the Security component, you can create firewalls that match certain request options.
-In most cases, matching against the URL is sufficient, but in special cases, you can further
-restrict the initialization of a firewall against other options of the request.
+When using the Security component, firewalls will decide whether they handle a request based on the
+result of a request matcher: the first firewall matching the request will handle it.
+
+The last firewall can be configured without any matcher to handle every incoming request.
+
+Restricting by Service
```

Comment:
I would change the order and first documention "Restricting by Config" and then "Restricting by Service". I think the service one is not used that much, whereas the configuration is almost always used.